docs/user_guide/supernode_ha.md

```diff
@@ -0,0 +1,127 @@
+# Supernode High Availability
+
+This doc contains all the design and usage document of supernode HA.
+
+## Two Goals Need to be Achieved
+
+To implement HA,we must guarantee the following two goals to be achieved:
+
+- Compatible with current mode
+- If a supernode is crash,other supernode can take over its work.In fact,supernode ha include two part,cdn file ha and memory information ha,In fact,this design only finishes cdn file ha.
+
+## Tool Introduction
+
+We can use distributed key-value store system such as etcd,zookeeper,consul and so on to manager the supernodes cluster.of course you can implement your own function to get supernodes information.
+now we use etcd to store every superondoe's information include peerID,IP and so on.
+
+Let’s focus on etcd because we will use it to construct our plan. etcd is a strongly consistent,distributed key-value store that provides a reliable way to store data that needs to
+be accessed by a distributed system or cluster of machines. It gracefully handles leader elections during network partitions and can tolerate machine failure, even in the leader node.
+
+### About Multi-Supernode Pattern
+
+Now we talk about how to implement file ha.In Dragonfly,we now has a multi-supernode pattern.This pattern can be a replace of ha,but it has these disadvantages,
+
+- Every supernode is stand-alone,so supernode can't communicate with each other for a better schedule.For example,pieces of a task are in different p2p network,so a supernode can only schedule its plan with its own peer instead of other peer which has pieces of a task in other p2p network.
+
+- Every supernode will download the same file repeatedly,which makes download slowly and a waste of disk place.This will be obvious when the supernode's num is large.
```

Comment:
"makes download slowly and a waste of disk place" -> makes download slow and wastes disk space.
"the supernode's num" -> the number of supernodes